### PR TITLE
Refactors animation presets ♻️

### PR DESCRIPTION
**Pull request by Enriqe**
Refactors animation preset by adding a set of common util functions for animations.

This in preparation of new animation presets that will be added for #12173.

/cc @newmuis

### CODE REVIEW

File: src/style.js [hunk 1]
```diff
@@ -175,6 +175,14 @@ export function px(value) {
   return value + 'px';
 }
 
+/**
+ * Returns a degree value.
+ * @param {number} value
+ * @return {string}
+ */
+export function deg(value) {
+  return value + 'deg';
```
Comment (alanorozco) [inline src/style.js]: nit: use interpolation.

File: src/style.js [hunk 2]
```diff
@@ -172,9 +172,17 @@ export function toggle(element, opt_display) {
...
-  return value + 'px';
+  return `${value}px`;
 }
 
+/**
+ * Returns a degree value.
+ * @param {number} value
+ * @return {string}
+ */
+export function deg(value) {
```
Comment (gmajoulet) [inline src/style.js]: Are you using it in your coming PRs? Otherwise, I don't think it's worth the abstraction

Comment (alanorozco) [reply src/style.js]: I think the readability benefit, plus AMP consistency make it worth keeping.

File: extensions/amp-story/0.1/animation-presets-utils.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ * @param {number} direction -1 for left, 1 for right
+ * @return {KeyframesDef} Keyframes that make up the animation.
+ */
+export function rotateAndTranslate(startX, startY, endX, endY, direction) {
+  return [
+    {transform: translate(startX, startY) + ' ' + rotate(direction * 360)},
+    {transform: translate(endX, endY) + ' ' + rotate(0)},
+  ];
+}
+
```
[the remark this answers is not in the document]
Comment (alanorozco) [reply extensions/amp-story/0.1/animation-presets-utils.js]: Top-level units should be split by two line-breaks per style guide.

File: src/style.js (hunk 2 above)
Comment (Enriqe) [reply src/style.js]: Leaving it for readability plus AMP consistency. Also highly likely that we will use it later for other animations.

File: extensions/amp-story/0.1/animation-presets-utils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Enriqe) [reply extensions/amp-story/0.1/animation-presets-utils.js]: Leaving it at one to stay consistent with the rest of AMP.

File: extensions/amp-story/0.1/animation-presets-utils.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import {KeyframesDef} from './animation-types';
+import {
+  rotate,
+  scale,
+  translate,
+} from '../../../src/style';
+
+/**
+ * Translates the element on the 2d plane according to the given points.
+ * @export
```
Comment (choumx) [inline extensions/amp-story/0.1/animation-presets-utils.js]: `@export` is unnecessary.

File: src/style.js (hunk 2 above)
Comment (newmuis) [reply src/style.js]: Also turns typos into compilation errors instead of just dropping the style, which is a good thing

**Pull request merged**